kratos/utilities/sparse_matrix_multiplication_utility.h

```diff
@@ -506,6 +506,90 @@ class SparseMatrixMultiplicationUtility
         delete[] aux_val_new_a;
     }
 
+    /**
+     * @brief This method computes of the transpose matrix of a given matrix
+     * @param rA The resulting matrix
+     * @param rB The second matrix to transpose
+     */
+    template <class AMatrix, class BMatrix>
+    static void TransposeMatrix(
+        AMatrix& rA,
+        const BMatrix& rB,
+        const double Factor = 1.0
+        )
+    {
+        // Get access to B data
+        const std::size_t* index1 = rB.index1_data().begin();
+        const std::size_t* index2 = rB.index2_data().begin();
+        const double* data = rB.value_data().begin();
+        const std::size_t transpose_nonzero_values = rB.value_data().end() - rB.value_data().begin();
+
+        const std::size_t size_system_1 = rB.size1();
+        const std::size_t size_system_2 = rB.size2();
+
+        if (rA.size1() != size_system_2 || rA.size2() != size_system_1 ) {
+            rA.resize(size_system_2, size_system_1);
+        }
+
+        std::size_t* new_a_ptr = new std::size_t[size_system_2 + 1];
```

Comment:
Can you use `vector` instead of `new`/`delete`?

Reply:
Technically this is more efficient, isn't it?, I mean the vector is for general purpose, this is just a c array for integers

Reply:
no reason for which it should be more efficient. DenseVector is only a tiny class, so here it should have no performance impact

Reply:
I can change it if that is a problem

Reply:
Practically, vector is just as efficient. In my implementation `v[n]` is equivalent to `*(this->_M_impl._M_start + n)`. But vector has a better interface and is less error prone so it is recommended to never use `new` / `delete` in newer c++ code, including performance critical code.

Reply:
The thing is that I pass an array of c for building the sparse matrix, because the sparse matrix is internally build that way

Reply:
one thing: do not use std::vector since std vector is initialized to zero. Use DenseVector instead. (That, DOES have a performance impact)

Reply:
The construction of the sparse matrix is done here https://github.com/KratosMultiphysics/Kratos/blob/eb0215f1dc21256d17523d65f96ad1780274c1c5/kratos/utilities/sparse_matrix_multiplication_utility.h#L600 I cannot used DenseVector without changing this, which means I need to change many things, both in this utilities and in the mixedulm_solver, which means many changes. can I do it in a different PR?

Reply:
I can

Reply:
It doesn't work

Reply:
For the std::vector I can, but I cannot for the DenseVector

Reply:
Both with data and with &v[0]

Reply:
Ok the problem is that it was taking DenseVector from some other include, on it. The next commit will be quite big, I inform